src/ds/non_blocking.h

```diff
@@ -102,10 +102,7 @@ namespace ringbuffer
       {
         for (auto& it : pending)
         {
-          const auto buffer_end = it.buffer.data() + it.buffer.size();
-          if (
-            it.marker == marker.value() &&
-            marker.value() != reinterpret_cast<uint64_t>(buffer_end))
+          if (it.marker == marker.value())
```

Comment:
If we try and write a 0-sized value though this, it would previously blow up and now works correctly. I believe we never have 0-sized writes in a real node, but could in some of the unit tests (randomly sized vectors of size 0). I've tried and failed to write a standalone unit test proving this is helpful, but it was found as part of this so slides in with this (for now).